### DISCUSSION TITLE
Made SEGA MARK3 to MD adapter and PCE adapter GERBER data for V3.

### DISCUSSION THREAD

**Discussion by Ryurron** [sanni/cartreader#345] [Look what I made]
SEGA MARK3 to MD adapter and PCE adapter GERBER data for V3.
Please use

[MARK3toMD_adapter_GERBER.zip](https://github.com/sanni/cartreader/files/8242598/MARK3toMD_adapter_GERBER.zip)
[PCE_adapter_GERBER.zip](https://github.com/sanni/cartreader/files/8242599/PCE_adapter_GERBER.zip)

**Comment by Bandmo** [comment]
Will this work for both pcengine and turbografx16 games?

**Comment by Ryurron** [comment]
Yes

**Comment by p0ojisan** [comment]
@Ryurron thanks. I ordered PCE adapter using your gerber file.

![IMG_2027](https://user-images.githubusercontent.com/95609164/165019624-c844753a-657a-4316-925b-cb00794c7eec.jpg)

Works correctly!

But in verification with my several Japanese Hu cards, I had to select "HuCARD Not Swapped" option. Plus, CRC never matches with the one in PCE_CRC_LIST.txt. 


![IMG_2030](https://user-images.githubusercontent.com/95609164/165020885-1136e4da-b3b6-43ab-9d3a-fa67d1e96208.jpg)

e.g.
:video_game: Youkai Douchuuki
CRC in txt: `F131B706`
Dumped data CRC: `80C3F824`


`80C3F824` is the correct one as I searched. And it works in emulator actually.

**Reply by Stanton54** [reply]
If anyone is having a hard time finding a slot you can get one at jt studios. It is just soldered on the pcb and works.
![image](https://user-images.githubusercontent.com/120297611/210705041-46eeaaae-bfdc-457b-99d5-62270257981c.jpeg)

https://jt-studios.com/product/38pin-pc-engine-connector/

**Reply by Cephiros** [reply]
I can second his recommendation, I got one a couple months ago and have had no problems with it.

**Comment by Ryurron** [comment]
@p0ojisan 
I am aware of the problem.
Where did you get the PCE pin slot?

**Reply by p0ojisan** [reply]
I got this slot from GAMEBANK-web.com adapter. 
https://www.amazon.co.jp/gp/product/B0826P6X3Q?language=en_US

Anyway, that is not hardware issue. So, it's fine and even can play with dumped rom.

**Comment by Ryurron** [comment]
Thanks for trying and reporting back😆

**Comment by sanni** [comment]
I updated the database to a newer no-intro set. Can you test if it works now?

**Reply by p0ojisan** [reply]
Hi @sanni, thank you for updating. Now it works correctly.

Tested these HuCards I have,
![IMG_2040](https://user-images.githubusercontent.com/95609164/165653221-1f0607bc-1ae2-480a-a245-b31d7fc68f20.jpg)

Yes, checksum is all good and renamed as expected.
![IMG_2037](https://user-images.githubusercontent.com/95609164/165653820-e134bad4-3e83-48f9-bc79-3414b75d153b.jpg)
![IMG_2038](https://user-images.githubusercontent.com/95609164/165653826-067b3329-c040-4940-9256-c532ed724275.jpg)
![IMG_2039](https://user-images.githubusercontent.com/95609164/165653828-a1782d77-0331-4a70-aa82-d1f48c26b691.jpg)


Again, thanks very much for such awesome project, Sanni, Ryurron and all of contributors.

**Comment by Ryurron** [comment]
wow
nice work!!
Thank you sanni, p0ojisan and I say Thank you for Save the hero.
I found out about cool projects thanks to it😆

**Comment by higashis** [comment]
@p0ojisan
Does it work correctly if you select "Turbochip" option instead of "HuCARD Not Swapped" option?

**Reply by higashis** [reply]
The pins from D0 to D7 are in the reverse order on this PCE adapter.
So, when you read a PC Engine HuCARD, you need to select the "Turbochip" option.
On the other hand, when you read a TurboGrafx-16 TurboChip, you need to select the "HuCARD" option.

**Reply by p0ojisan** [reply]
yep, also works with "Turbochip" option.

**Reply by higashis** [reply]
I fixed it.
Please use

[PCE_adapter_GERBER_2.zip](https://github.com/sanni/cartreader/files/8596209/PCE_adapter_GERBER_2.zip)

**Comment by KytorIndustries** [comment]
These just arrived from PCBway and look great. I have some extras I'm not going to need. Do you have any issues with me selling these or giving them away?

**Reply by higashis** [reply]
Please do not sell on save the hero project.

**Reply by SAVE-THE-HERO** [reply]
Hello
I did not ask permission from Mr. higashis  because I was using the data that Mr.Ryurron modified for me.
Sorry about that.
Do I need your permission too?
If so, I will stop the sale immediately.

**Reply by higashis** [reply]
Please do not sell on save the hero project.

**Reply by kamalawala** [reply]
I kinda hoped that, like the cart reader itself, custom adapters would be licensed under CC-BY-4.0.

**Reply by higashis** [reply]
SAVE-THE-HERO is lying.
Look at the patterns around 20.
This is my data.
![pce_adapter](https://user-images.githubusercontent.com/102914562/185974360-25b7a718-2a34-4a60-b7ff-e6fe94bf4f31.jpg)

**Reply by Oswalds99** [reply]
Do you have any issues with me selling these or giving them away?
I dont selll in Japan and I am not Save the hero.

**Reply by SAVE-THE-HERO** [reply]
Hello
To eliminate problems, we have created a new PCE adapter.
On the back side is a note that reads
 "Created by modifying 
[Open Source Cartridge Reader]
sanni - https://github.com/sanni/cartreader
Licensed under CC BY 4.0”
Licence on the back.

*Untested
[PCE_GERBER_220823.zip](https://github.com/sanni/cartreader/files/9400217/PCE_GERBER_220823.zip)
<img width="805" alt="ss01" src="https://user-images.githubusercontent.com/89830526/186096288-5b2f745f-606b-4977-8d16-3e304c031087.png">
<img width="800" alt="ss02" src="https://user-images.githubusercontent.com/89830526/186096297-8a3346f1-69df-49c9-8cf4-23d7fb27a8e0.png">
<img width="404" alt="ss03" src="https://user-images.githubusercontent.com/89830526/186096305-6a70d8e6-0e0a-4d9e-89c9-9a9d88c9acb4.png">

**Reply by kamalawala** [reply]
![Epic win!](https://user-images.githubusercontent.com/111480334/186099966-10c8d289-6102-4646-98a9-3a686c99ee47.gif)

**Reply by SAVE-THE-HERO** [reply]
PCE adapter has been tested
OK now

**Comment by PW5190** [comment]
I have several questions about the MARK III adapter: Can it dump SG-1000/SC-3000 cartridges (I know the JP Master System and MARK III can play SG-1000/SC-3000 games without the fuss via back compat.) as well as SEGA MyCard games using the Card Catcher (I know SEGA made this accessory specifically for the SG-1000/SC-3000 to play MyCard games through its only cartridge slot, but is not needed for the MARK III and Master System, since the card slot was built-in.)?

**Reply by PsyK0p4T** [reply]
Months later, I can answer to this.
So yes, a Mark III adapter is able to dump SG-1000 cartridges, by selecting "SG-1000 raphnet" in OSCR menu. 
Just at the current state, there is no rom check possibility after dumping process. It would be cool if someone could make a "SG1000.txt" on the same model as "SMS.txt". I have the stuff here to test it and merge it to the project then.

**Reply by sanni** [reply]
I have now added a SG-1000 database.

**Reply by PsyK0p4T** [reply]
damn that was fast ^^ thanks sanni

**Comment by Mythris29** [comment]
Any chance of getting an assembly guide for the Mark III adapter? and/or a parts list? Thanks!

**Reply by SAVE-THE-HERO** [reply]
Bridge to R3 and R4

Slot is here
https://www.digikey.jp/ja/products/detail/edac-inc/396-044-520-201/1297924?utm_medium=email&utm_source=oce&utm_campaign=3316_OCE22RT&utm_content=productdetail_JP&utm_cid=2802144&so=76545912&mkt_tok=MDI4LVNYSy01MDcAAAGFQxxViMlmh6mbbU2YwQS7dZgfHL5egzbBGVlY3cjt9bseYDIAa051M80cxCoJZbaynizUeybTNAqTPloGgW8Fa2Rj7Ynk28z30g0cHif8

**Reply by tslpre** [reply]
that is really useful, thank you - what about the "R" parts? It seems that those are meant to host a resistor each? If yes, which resistors should be used?

**Comment by PW5190** [comment]
While on topic of adapters, will there also be adapters for dumping Atari cartridges (2600/7800, 5200, Jaguar, Lynx)?

**Comment by PW5190** [comment]
And while we are talking about SEGA in general, can the Cartridge Reader dump 32X cartridge games? If not, will there be a new update for it? 32X games could be inserted directly into the Genesis/Mega Drive cart slot, since the 32X cartridge pinout is similar to the Genesis/Mega Drive's.

**Reply by Cephiros** [reply]
I find, when read in mega drive mode, 32x dumps just fine. I will note however that it does not seem to reference 32x.txt, so it will not validate. I have crcd the resulting rom manually and can match it to entries in 32x. I suspect if you merged md.txt and 32x.txt into one file, md.txt, it will find them, but i never cared enough to try.

**Reply by sanni** [reply]
This should be fixed in V10.3

**Comment by PW5190** [comment]
Can you use Sonic & Knuckles with the OSCR to dump its lock-on games?

Sonic & Knuckles by itself with no cartridge is just the original Sonic & Knuckles.
Sonic & Knuckles with Sonic 1 attached is the full Blue Sphere game.
Sonic & Knuckles with Sonic 2 attached is Knuckles in Sonic 2, where you can play Sonic 2 as Knuckles.
Sonic & Knuckles with Sonic 3 attached is the full Sonic 3 & Knuckles experience.
Sonic & Knuckles with any other cartridge attached can depend whether how much memory is in the cartridge. (Smaller memory can launch Blue Sphere partially, depending on the cart, while larger memory just launch Sonic & Knuckles as if nothing is attached.)

**Reply by scrap-a** [reply]
That was supported last year, but now I tried the latest version ( Firm V11.2 / Hard V3), it was NG in my environment.
Apparently, the compiler's behavior for variable initialization has changed, and it no longer works.
We will fix it soon. If possible, please check the operation after the fix.

(By the way, why did you ask the Sonic & Knuckles question in this thread?)

**Reply by PW5190** [reply]
> By the way, why did you ask the Sonic & Knuckles question in this thread?

I just wanted to ask. If you want, I can move it to another discussion.

**Reply by scrap-a** [reply]
That fix has already been merged. You can try it if you like.

> I just wanted to ask. If you want, I can move it to another discussion.

No need to move it.
This thread's title is "Made SEGA MARK3 to MD adapter and PCE adapter GERBER data for V3.", so your question that is not related this is hard to find, I just thought.

**Comment by partlyhuman** [comment]
It should be noted here that you need to fabricate these adapters with a PCB thickness of *1.2mm*. Please correct me if I'm wrong about that.

**Reply by tslpre** [reply]
I ordered a MarkIII to MD adapter a few days ago with a thickness of 1.6 and it works without an issue. Inserting/removing it is easy, I don't see that this thickness might been forcing the connector.

**Comment by PW5190** [comment]
I want to ask a question. Could you use SEGA's official Power Base Converter for Genesis (or Mega Adaptor for Mega Drive) to dump Master System/Mark III/SG-1000 cartridges and cards? The accessory features both cartridge and card slots.
